### ISSUE TITLE
Update terraform provider for AKSv2

### ISSUE THREAD

**Issue by gaktive** [rancher/rancher#31296]
We need to update the terraform operator for the AKS v2 operator: https://github.com/rancher/aks-operator

**Comment by sowmyav27** [comment]
To validate:
- Test deploying AKSv2 through TF with minimal input parameters required set to bring up the cluster
- Test deploying AKSv2 through TF with all applicable input parameters set to bring up the cluster

**Comment by rawmind0** [comment]
@sowmyav27, i'd also add, Test importing AKSv2 through TF

**Comment by rawmind0** [comment]
Binaries to test https://github.com/rancher/terraform-provider-rancher2/pull/711#issuecomment-911546295

**Comment by izaac** [comment]
Issue found when using the azure network_plugin https://github.com/rancher/rancher/issues/34750

**Comment by izaac** [comment]
Issue found when using the default node pool example https://github.com/rancher/rancher/issues/34752

**Comment by izaac** [comment]
Rancher version: `v2.6-head (09/20/2021)` _1f15852d_
Rancher version: `master-head (09/20/2021)` _452be57_

I was able to do AKSv2 Create and Import using all optional and the minimum required parameters.
It's important to know that depending on the `network_plugin` parameter value there are additional CIDR parameters that can be set but are optional when using `network_plugin=azure`. This is when creating not importing.

- I was able to create/import 
- Edit/Update
- Destroy

Some issues were found and opened as separate for tracking and linked in this issue.
